R/optparse.R

```diff
@@ -485,7 +491,26 @@ parse_args <- function(object, args = commandArgs(trailingOnly = TRUE),
                 options_list[[option@dest]] <- FALSE
             } else {    
                 options_list[[option@dest]] <- option_value
+            }         
+            if (option@action == "callback") {
+                if(!is.function(option@callback))
+                    stop(springf("callback not callable"))
+                if(option@callback_args != NULL && (!is.list(callback_args) && names(callback_args) != NULL)) 
+                    stop(sprinf("callback_args, if supplied, must be ordinary list"))
+                if(option@callback_kwargs != NULL && (!is.list(callback_kwargs) && names(callback_args) == NULL))
+                    stop(sprinf("callback_kwargs, if supplied, must be ordinary list"))
+
+                options_list[[option@dest]] <- option@callback(option, option_value, object, args, kwargs)
```

Comment:
Between ``option`` and ``option_value`` there should be an ``opt_str`` argument:

    options_list[[option@dest]] <- option@callback(option, option$long_flag, option_value, object, args, kwargs)